terraform/autoneg/main.tf

```diff
@@ -0,0 +1,335 @@
+    resources  = ["services/status"]
+    verbs      = ["get", "patch", "update"]
+  }
+}
+
+resource "kubernetes_cluster_role" "clusterrole_autoneg_proxy_role" {
+  metadata {
+    name = "autoneg-proxy-role"
+
+    labels = {
+      app = "autoneg"
+    }
+    annotations = {}
+  }
+
+  rule {
+    api_groups = ["authentication.k8s.io"]
+    resources  = ["tokenreviews"]
+    verbs      = ["create"]
+  }
+
+  rule {
+    api_groups = ["authorization.k8s.io"]
+    resources  = ["subjectaccessreviews"]
+    verbs      = ["create"]
+  }
+}
+
+resource "kubernetes_role_binding" "rolebinding_autoneg_leader_election_rolebinding" {
+  metadata {
+    namespace = kubernetes_namespace.namespace_autoneg_system.metadata[0].name
+    name      = "autoneg-leader-election-rolebinding"
+
+    labels = {
+      app = "autoneg"
+    }
+    annotations = {}
+  }
+
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "Role"
+    name      = kubernetes_role.role_autoneg_leader_election_role.metadata[0].name
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = kubernetes_service_account.service_account.metadata[0].name
+    namespace = kubernetes_namespace.namespace_autoneg_system.metadata[0].name
+  }
+}
+
+resource "kubernetes_cluster_role_binding" "clusterrolebinding_autoneg_manager_rolebinding" {
+  metadata {
+    name = "autoneg-manager-rolebinding"
+    labels = {
+      app = "autoneg"
+    }
+    annotations = {}
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = kubernetes_cluster_role.clusterrole_autoneg_manager_role.metadata[0].name
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = kubernetes_service_account.service_account.metadata[0].name
+    namespace = kubernetes_namespace.namespace_autoneg_system.metadata[0].name
+  }
+}
+
+resource "kubernetes_cluster_role_binding" "clusterrolebinding_autoneg_proxy_rolebinding" {
+  metadata {
+    name = "autoneg-proxy-rolebinding"
+    labels = {
+      app = "autoneg"
+    }
+    annotations = {}
+  }
+  role_ref {
+    api_group = "rbac.authorization.k8s.io"
+    kind      = "ClusterRole"
+    name      = kubernetes_cluster_role.clusterrole_autoneg_proxy_role.metadata[0].name
+  }
+  subject {
+    kind      = "ServiceAccount"
+    name      = kubernetes_service_account.service_account.metadata[0].name
+    namespace = kubernetes_namespace.namespace_autoneg_system.metadata[0].name
+  }
+}
+
+resource "kubernetes_service" "service_autoneg_controller_manager_metrics_service" {
+  metadata {
+    annotations = {
+      "prometheus.io/port"   = "8443"
+      "prometheus.io/scheme" = "https"
+      "prometheus.io/scrape" = "true"
+      "cloud.google.com"     = "'{\"ingress\":true}'"
+    }
+    labels = {
+      "app"           = "autoneg"
+      "control-plane" = "controller-manager"
+    }
+    name      = "autoneg-controller-manager-metrics-service"
+    namespace = "autoneg-system"
```

Comment:
nit: `namespace = kubernetes_namespace.namespace.metadata[0].name` would avoid the `depends_on` below

Reply:
Fixed.